File: assets/sources/example1.go
// example1.go

package main

import (
	"fmt"
	"math/rand/v2"
	"os"
	"strconv"
)

func main() {
	inside := 0
	n := 100_000_000
	if len(os.Args) > 1 {
		n, _ = strconv.Atoi(os.Args[1])
	}
	for range n {
		x := rand.Float64()
		y := rand.Float64()
		if x*x+y*y <= 1 {
			inside++
		}
	}
	fmt.Println(float64(inside) * 4 / float64(n))
}

File: assets/sources/example1.jl
# montecarlo.jl

is_inside(x ::Float64, y ::Float64) = x*x + y*y <= 1.0

function calculate(iterations ::Int)
    inside = 1
    for i in 1:iterations
        if is_inside(rand(), rand())
            inside += 1
        end
    end
    return inside * 4.0 / iterations
end

print(calculate(100_000_000))
